// inferred-types@0.50.11/dist/types/string-literals/character-sets/images/Exif.js
export var ExifCompression;
(function (ExifCompression) {
    ExifCompression[ExifCompression["Uncompressed"] = 1] = "Uncompressed";
    ExifCompression[ExifCompression["CCITT"] = 2] = "CCITT";
    ExifCompression[ExifCompression["T4Group3Fax"] = 3] = "T4Group3Fax";
    ExifCompression[ExifCompression["T6Group3Fax"] = 4] = "T6Group3Fax";
    ExifCompression[ExifCompression["LZW"] = 5] = "LZW";
    ExifCompression[ExifCompression["JpgOldStyle"] = 6] = "JpgOldStyle";
    ExifCompression[ExifCompression["Jpg"] = 7] = "Jpg";
    ExifCompression[ExifCompression["AdobeDeflate"] = 8] = "AdobeDeflate";
    ExifCompression[ExifCompression["JBigBw"] = 9] = "JBigBw";
    ExifCompression[ExifCompression["JBigColor"] = 10] = "JBigColor";
    ExifCompression[ExifCompression["JpegAlt"] = 99] = "JpegAlt";
    ExifCompression[ExifCompression["Kodak262"] = 262] = "Kodak262";
    ExifCompression[ExifCompression["Next"] = 32766] = "Next";
    ExifCompression[ExifCompression["SonyRawCompressed"] = 32767] = "SonyRawCompressed";
    ExifCompression[ExifCompression["PackedRaw"] = 32769] = "PackedRaw";
    ExifCompression[ExifCompression["SamsungSrwCompressed"] = 32770] = "SamsungSrwCompressed";
    ExifCompression[ExifCompression["CCIRLEW"] = 32771] = "CCIRLEW";
    ExifCompression[ExifCompression["SamsungSrwCompressed2"] = 32772] = "SamsungSrwCompressed2";
    ExifCompression[ExifCompression["Packbits"] = 32773] = "Packbits";
    ExifCompression[ExifCompression["Thunderscan"] = 32809] = "Thunderscan";
    ExifCompression[ExifCompression["KodakKdcCompressed"] = 32867] = "KodakKdcCompressed";
    ExifCompression[ExifCompression["IT8CTPAD"] = 32895] = "IT8CTPAD";
    ExifCompression[ExifCompression["IT8LW"] = 32896] = "IT8LW";
    ExifCompression[ExifCompression["IT8MP"] = 32897] = "IT8MP";
    ExifCompression[ExifCompression["IT8BL"] = 32898] = "IT8BL";
    ExifCompression[ExifCompression["PixarFilm"] = 32908] = "PixarFilm";
    ExifCompression[ExifCompression["PixarLog"] = 32909] = "PixarLog";
    ExifCompression[ExifCompression["Deflate"] = 32946] = "Deflate";
    ExifCompression[ExifCompression["DCS"] = 32947] = "DCS";
    ExifCompression[ExifCompression["AperioJpeg2000YCbCr"] = 33003] = "AperioJpeg2000YCbCr";
    ExifCompression[ExifCompression["AperioJpeg2000RGB"] = 33005] = "AperioJpeg2000RGB";
    ExifCompression[ExifCompression["JBig"] = 34661] = "JBig";
    ExifCompression[ExifCompression["SGILog"] = 34676] = "SGILog";
    ExifCompression[ExifCompression["SGILog24"] = 34677] = "SGILog24";
    ExifCompression[ExifCompression["Jpeg2000"] = 34712] = "Jpeg2000";
    ExifCompression[ExifCompression["NikonNEFCompressed"] = 34713] = "NikonNEFCompressed";
    ExifCompression[ExifCompression["JBig2TiffFx"] = 34715] = "JBig2TiffFx";
    ExifCompression[ExifCompression["MicrosoftBinaryLevelCodec"] = 34718] = "MicrosoftBinaryLevelCodec";
    ExifCompression[ExifCompression["MicrosoftProgressiveTransformCodec"] = 34719] = "MicrosoftProgressiveTransformCodec";
    ExifCompression[ExifCompression["MicrosoftVector"] = 34720] = "MicrosoftVector";
    ExifCompression[ExifCompression["ESRCLerc"] = 34887] = "ESRCLerc";
    ExifCompression[ExifCompression["LossyJpeg"] = 34892] = "LossyJpeg";
    ExifCompression[ExifCompression["LZMA2"] = 34925] = "LZMA2";
    ExifCompression[ExifCompression["Zstd"] = 34926] = "Zstd";
    ExifCompression[ExifCompression["WepP"] = 34927] = "WepP";
    ExifCompression[ExifCompression["PNG"] = 34933] = "PNG";
    ExifCompression[ExifCompression["JpegXR"] = 34934] = "JpegXR";
    ExifCompression[ExifCompression["KodakDCRCompressed"] = 65000] = "KodakDCRCompressed";
    ExifCompression[ExifCompression["PentaxPEFCompressed"] = 65535] = "PentaxPEFCompressed";
})(ExifCompression || (ExifCompression = {}));
export var ExifLightSource;
(function (ExifLightSource) {
    ExifLightSource[ExifLightSource["Unknown"] = 0] = "Unknown";
    ExifLightSource[ExifLightSource["Daylight"] = 1] = "Daylight";
    ExifLightSource[ExifLightSource["Fluorescent"] = 2] = "Fluorescent";
    ExifLightSource[ExifLightSource["Tungsten"] = 3] = "Tungsten";
    ExifLightSource[ExifLightSource["Flash"] = 4] = "Flash";
    ExifLightSource[ExifLightSource["FineWeather"] = 5] = "FineWeather";
    ExifLightSource[ExifLightSource["Cloudy"] = 6] = "Cloudy";
    ExifLightSource[ExifLightSource["Shade"] = 7] = "Shade";
    ExifLightSource[ExifLightSource["DaylightFluorescent"] = 8] = "DaylightFluorescent";
    ExifLightSource[ExifLightSource["DayWhiteFluorescent"] = 9] = "DayWhiteFluorescent";
    ExifLightSource[ExifLightSource["CoolWhiteFluorescent"] = 10] = "CoolWhiteFluorescent";
    ExifLightSource[ExifLightSource["WhiteFluorescent"] = 11] = "WhiteFluorescent";
    ExifLightSource[ExifLightSource["WarmWhiteFluorescent"] = 12] = "WarmWhiteFluorescent";
    ExifLightSource[ExifLightSource["StandardLightA"] = 13] = "StandardLightA";
    ExifLightSource[ExifLightSource["StandardLightB"] = 14] = "StandardLightB";
    ExifLightSource[ExifLightSource["StandardLightC"] = 15] = "StandardLightC";
    ExifLightSource[ExifLightSource["D55"] = 16] = "D55";
    ExifLightSource[ExifLightSource["D65"] = 17] = "D65";
    ExifLightSource[ExifLightSource["D75"] = 18] = "D75";
    ExifLightSource[ExifLightSource["D50"] = 19] = "D50";
    ExifLightSource[ExifLightSource["ISOStudioTungsten"] = 20] = "ISOStudioTungsten";
    ExifLightSource[ExifLightSource["Other"] = 255] = "Other";
})(ExifLightSource || (ExifLightSource = {}));
export var ExifFlashValues;
(function (ExifFlashValues) {
    ExifFlashValues[ExifFlashValues["NoFlash"] = 0] = "NoFlash";
    ExifFlashValues[ExifFlashValues["Fired"] = 1] = "Fired";
    ExifFlashValues[ExifFlashValues["FiredReturnNotDetected"] = 5] = "FiredReturnNotDetected";
    ExifFlashValues[ExifFlashValues["FiredReturnDetected"] = 7] = "FiredReturnDetected";
    ExifFlashValues[ExifFlashValues["OnDidNotFire"] = 8] = "OnDidNotFire";
    ExifFlashValues[ExifFlashValues["OnFired"] = 9] = "OnFired";
    ExifFlashValues[ExifFlashValues["OnReturnNotDetected"] = 13] = "OnReturnNotDetected";
    ExifFlashValues[ExifFlashValues["OnReturnDetected"] = 15] = "OnReturnDetected";
    ExifFlashValues[ExifFlashValues["OffDidNotFire"] = 16] = "OffDidNotFire";
    ExifFlashValues[ExifFlashValues["OffDidNotFireReturnNotDetected"] = 20] = "OffDidNotFireReturnNotDetected";
    ExifFlashValues[ExifFlashValues["AutoDidNotFire"] = 24] = "AutoDidNotFire";
    ExifFlashValues[ExifFlashValues["AutoFired"] = 25] = "AutoFired";
    ExifFlashValues[ExifFlashValues["AutoFiredReturnNotDetected"] = 29] = "AutoFiredReturnNotDetected";
    ExifFlashValues[ExifFlashValues["AutoFiredReturnDetected"] = 31] = "AutoFiredReturnDetected";
    ExifFlashValues[ExifFlashValues["NoFlashFunction"] = 32] = "NoFlashFunction";
    ExifFlashValues[ExifFlashValues["OffNoFlashFunction"] = 48] = "OffNoFlashFunction";
    ExifFlashValues[ExifFlashValues["FiredRedEyeReduction"] = 65] = "FiredRedEyeReduction";
    ExifFlashValues[ExifFlashValues["FiredRedEyeReductionReturnNotDetected"] = 69] = "FiredRedEyeReductionReturnNotDetected";
    ExifFlashValues[ExifFlashValues["FiredRedEyeReductionReturnDetected"] = 71] = "FiredRedEyeReductionReturnDetected";
    ExifFlashValues[ExifFlashValues["OnRedEyeReduction"] = 73] = "OnRedEyeReduction";
    ExifFlashValues[ExifFlashValues["OnRedEyeReductionReturnNotDetected"] = 77] = "OnRedEyeReductionReturnNotDetected";
    ExifFlashValues[ExifFlashValues["OnRedEyeReductionReturnDetected"] = 79] = "OnRedEyeReductionReturnDetected";
    ExifFlashValues[ExifFlashValues["OffRedEyeReduction"] = 80] = "OffRedEyeReduction";
    ExifFlashValues[ExifFlashValues["AutoDidNotFireRedEyeReduction"] = 88] = "AutoDidNotFireRedEyeReduction";
    ExifFlashValues[ExifFlashValues["AutoFiredRedEyeReduction"] = 89] = "AutoFiredRedEyeReduction";
    ExifFlashValues[ExifFlashValues["AutoFiredRedEyeReductionNotDetected"] = 96] = "AutoFiredRedEyeReductionNotDetected";
    ExifFlashValues[ExifFlashValues["AutoFiredRedEyeReductionDetected"] = 93] = "AutoFiredRedEyeReductionDetected";
})(ExifFlashValues || (ExifFlashValues = {}));
export var ExifPreviewColorSpace;
(function (ExifPreviewColorSpace) {
    ExifPreviewColorSpace[ExifPreviewColorSpace["Unknown"] = 0] = "Unknown";
    ExifPreviewColorSpace[ExifPreviewColorSpace["GrayGamma22"] = 1] = "GrayGamma22";
    ExifPreviewColorSpace[ExifPreviewColorSpace["sRGB"] = 2] = "sRGB";
    ExifPreviewColorSpace[ExifPreviewColorSpace["AdobeRGB"] = 3] = "AdobeRGB";
    ExifPreviewColorSpace[ExifPreviewColorSpace["ProPhotoRGB"] = 4] = "ProPhotoRGB";
})(ExifPreviewColorSpace || (ExifPreviewColorSpace = {}));
export var ExifEmbedPolicy;
(function (ExifEmbedPolicy) {
    ExifEmbedPolicy[ExifEmbedPolicy["AllowCopying"] = 0] = "AllowCopying";
    ExifEmbedPolicy[ExifEmbedPolicy["EmbedIfUsed"] = 1] = "EmbedIfUsed";
    ExifEmbedPolicy[ExifEmbedPolicy["NeverEmbed"] = 2] = "NeverEmbed";
    ExifEmbedPolicy[ExifEmbedPolicy["NoRestrictions"] = 3] = "NoRestrictions";
})(ExifEmbedPolicy || (ExifEmbedPolicy = {}));
export var ExifSubjectDistance;
(function (ExifSubjectDistance) {
    ExifSubjectDistance[ExifSubjectDistance["Unknown"] = 0] = "Unknown";
    ExifSubjectDistance[ExifSubjectDistance["Macro"] = 1] = "Macro";
    ExifSubjectDistance[ExifSubjectDistance["Close"] = 2] = "Close";
    ExifSubjectDistance[ExifSubjectDistance["Distant"] = 3] = "Distant";
})(ExifSubjectDistance || (ExifSubjectDistance = {}));
export var ExifSharpness;
(function (ExifSharpness) {
    ExifSharpness[ExifSharpness["Normal"] = 0] = "Normal";
    ExifSharpness[ExifSharpness["Soft"] = 1] = "Soft";
    ExifSharpness[ExifSharpness["Hard"] = 2] = "Hard";
})(ExifSharpness || (ExifSharpness = {}));
export var ExifSceneCaptureType;
(function (ExifSceneCaptureType) {
    ExifSceneCaptureType[ExifSceneCaptureType["Standard"] = 0] = "Standard";
    ExifSceneCaptureType[ExifSceneCaptureType["Landscape"] = 1] = "Landscape";
    ExifSceneCaptureType[ExifSceneCaptureType["Portrait"] = 2] = "Portrait";
    ExifSceneCaptureType[ExifSceneCaptureType["Night"] = 3] = "Night";
    ExifSceneCaptureType[ExifSceneCaptureType["Other"] = 4] = "Other";
})(ExifSceneCaptureType || (ExifSceneCaptureType = {}));
export var ExifGainControl;
(function (ExifGainControl) {
    ExifGainControl[ExifGainControl["None"] = 0] = "None";
    ExifGainControl[ExifGainControl["LowGainUp"] = 1] = "LowGainUp";
    ExifGainControl[ExifGainControl["HighGainUp"] = 2] = "HighGainUp";
    ExifGainControl[ExifGainControl["LowGainDown"] = 3] = "LowGainDown";
    ExifGainControl[ExifGainControl["HighGainDown"] = 4] = "HighGainDown";
})(ExifGainControl || (ExifGainControl = {}));
export var ExifContrast;
(function (ExifContrast) {
    ExifContrast[ExifContrast["Normal"] = 0] = "Normal";
    ExifContrast[ExifContrast["Low"] = 1] = "Low";
    ExifContrast[ExifContrast["High"] = 2] = "High";
})(ExifContrast || (ExifContrast = {}));
export var ExifSaturation;
(function (ExifSaturation) {
    ExifSaturation[ExifSaturation["Normal"] = 0] = "Normal";
    ExifSaturation[ExifSaturation["Low"] = 1] = "Low";
    ExifSaturation[ExifSaturation["High"] = 2] = "High";
})(ExifSaturation || (ExifSaturation = {}));
