// inferred-types@0.50.8/dist/constants/Retail.js
/**
 * Amazon's DNS entries (for retail business)
 */
export const AMAZON_DNS = [
    "www.amazon.com",
    `www.amazon.co.uk`,
    `www.amazon.fr`,
    `www.amazon.de`,
    `www.amazon.it`,
    `www.amazon.es`
];
export const WALMART_DNS = [
    "www.walmart.com",
    "www.walmart.ca",
    "www.walmart.com.mx"
];
export const APPLE_DNS = [
    "store.apple.com",
    "www.apple.com",
];
export const HOME_DEPOT_DNS = [
    "www.homedepot.com",
    "www.homedepot.ca",
];
export const EBAY_DNS = [
    "www.ebay.com",
    "www.ebay.co.uk",
    "www.ebay.fr",
    "www.ebay.de",
    "www.ebay.it"
];
export const TARGET_DNS = [
    "www.target.com"
];
export const ETSY_DNS = [
    "www.etsy.com"
];
export const COSTCO_DNS = [
    "www.costco.com",
    "www.costco.co.uk",
    "www.costco.ca"
];
export const CHEWY_DNS = [
    "www.chewy.com"
];
export const BEST_BUY_DNS = [
    "www.bestbuy.com",
    "www.bestbuy.ca"
];
export const WAYFAIR_DNS = [
    "www.wayfair.com",
    "www.wayfair.ca",
    "www.wayfair.co.uk",
];
export const NIKE_DNS = [
    "www.nike.com"
];
export const MACYS_DNS = [
    "www.macys.com"
];
export const LOWES_DNS = [
    "www.lowes.com"
];
export const IKEA_DNS = [
    "www.ikea.com"
];
export const DELL_DNS = [
    "www.dell.com"
];
/** DNS for H&M retailer */
export const HM_DNS = [
    "www2.hm.com"
];
export const ZARA_DNS = [
    "www.zara.com"
];
export const KROGER_DNS = [
    "www.kroger.com"
];
export const WALGREENS_DNS = [
    "www.walgreens.com"
];
export const CVS_DNS = [
    "www.cvs.com"
];
export const WHOLE_FOODS_DNS = [
    "www.wholefoodsmarket.com",
    "www.wholefoodsmarket.co.uk",
];
